pom.xml

```diff
@@ -72,65 +72,126 @@
+                    <workingDirectory>${project.build.directory}/src</workingDirectory>
+                    <arguments>
+                        <argument>${revision}</argument>
+                        <argument>${project.build.directory}/src</argument>
+                    </arguments>
                 </configuration>
                 <executions>
                     <execution>
-                        <id>build-docker-image</id>
-                        <phase>initialize</phase>
+                        <id>docker-generate-binaries</id>
+                        <phase>compile</phase>
                         <goals>
-                            <goal>build</goal>
+                            <goal>exec</goal>
                         </goals>
                     </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>3.2.0</version>
+                <executions>
                     <execution>
-                        <id>compile-in-docker</id>
-                        <phase>compile</phase>
+                        <id>copy-source</id>
+                        <phase>generate-sources</phase>
                         <goals>
-                            <goal>start</goal>
+                            <goal>copy-resources</goal>
                         </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}</outputDirectory>
+                            <overwrite>true</overwrite>
+                            <resources>
+                                <resource>
+                                    <directory>${project.basedir}</directory>
+                                    <includes>
+                                        <include>src/**</include>
+                                    </includes>
+                                </resource>
+                            </resources>
+                        </configuration>
                     </execution>
                     <execution>
-                        <id>docker-cleanup</id>
-                        <phase>package</phase>
+                        <id>copy-test-results</id>
+                        <phase>test</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/surefire-reports/</outputDirectory>
+                            <overwrite>true</overwrite>
+                            <resources>
+                                <resource>
+                                    <directory>${project.build.directory}/src/test-results</directory>
```

Comment:
`src/test-results`?!

Reply:
yeah, so this is definitely all sorts of convoluted, which is why it's this mess:
1. maven copies the `src` dir into `target` - because I didn't want to keep modifying the `src` dir and showing all these SCM changes
2. `docker run` will also mount that copied `src` dir (ie. `target/src`)
3. after i generated the test results, they would be left in the mounted dir, hence `target/src/test-results`

So yeah, using your suggestion from https://github.com/jenkinsci/lib-durable-task/pull/6#discussion_r623846569 I'm going to get rid of all this nastiness

Reply:
removed all of this file moving and docker mounting by using `docker build --output` for linux, and `docker create` for windows in 48efa4fe70f4ba2b9a29944253e7934a4afd8406